contracts/core/OTCWrapper.sol

```diff
@@ -0,0 +1,746 @@
+     *  OTC OPERATIONS
+     ***********************************************/
+
+    /**
+     * @notice places an order
+     * @param _underlying underlying asset address
+     * @param _isPut option type the vault is selling
+     * @param _strikePrice option strike price (with its respective token decimals)
+     * @param _expiry option expiry timestamp
+     * @param _premium order premium amount (USDC value with USDC decimals)
+     * @param _notional order notional (USD value with 6 decimals)
+     */
+    function placeOrder(
+        address _underlying,
+        bool _isPut,
+        uint256 _strikePrice,
+        uint256 _expiry,
+        uint256 _premium,
+        uint256 _notional
+    ) external {
+        require(
+            _notional > minMaxNotional[_underlying].min && _notional < minMaxNotional[_underlying].max,
+            "OTCWrapper: invalid notional value"
+        );
+        require(_expiry > block.timestamp, "OTCWrapper: expiry must be in the future");
+
+        latestOrder += 1;
+
+        orders[latestOrder] = Order(
+            _underlying,
+            address(0),
+            _isPut,
+            _strikePrice,
+            _expiry,
+            _premium,
+            _notional,
+            _msgSender(),
+            address(0),
+            0,
+            address(0),
+            block.timestamp
+        );
+
+        ordersByAcct[_msgSender()].push(latestOrder);
+
+        orderStatus[latestOrder] = OrderStatus.Pending;
+
+        emit OrderPlaced(latestOrder, _underlying, _isPut, _strikePrice, _expiry, _premium, _notional, _msgSender());
+    }
+
+    /**
+     * @notice cancels an order
+     * @param _orderID order id
+     */
+    function undoOrder(uint256 _orderID) external {
+        require(orderStatus[_orderID] == OrderStatus.Pending, "OTCWrapper: inexistent or unsuccessful order");
+        require(orders[_orderID].buyer == _msgSender(), "OTCWrapper: only buyer can undo the order");
+
+        orderStatus[_orderID] = OrderStatus.Failed;
+
+        emit OrderCancelled(_orderID);
+    }
+
+    /**
+     * @notice executes an order
+     * @dev can only be called by whitelisted market makers
+     *      requires that product and collateral have already been whitelisted beforehand
+     *      ensure collateral naked cap from Controller.sol is high enough for the additional collateral
+     *      ensure setUpperBoundValues and setSpotShock from MarginCalculator.sol have been set up
+     * @param _orderID id of the order
+     * @param _userSignature user permit signature
+     * @param _mmSignature market maker permit signature
+     * @param _premium order premium amount (USDC value with USDC decimals)
+     * @param _collateralAsset collateral asset address
+     * @param _collateralAmount collateral amount (with its respective token decimals)
+     */
+    function executeOrder(
+        uint256 _orderID,
+        Permit calldata _userSignature,
```

Comment:
lets add 2 decimals for fees as you said

Reply:
https://github.com/ribbon-finance/GammaProtocol-OTC/pull/2/commits/9ff2d61f014d96dcae3f2327e91c92d32d02dfaa